### DISCUSSION TITLE
CFDP Backwards Compatibility

### DISCUSSION THREAD

**Discussion by space-individual** [nasa/cFS#934] [Q&A]
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/CF/blob/main/CONTRIBUTING.md).
* [x] I reviewed the [CF README.md file](https://github.com/nasa/CF/blob/main/README.md) to see if the feature is in the major future work.
* [x] I performed a cursory search to see if the feature request is relevant, not redundant, nor in conflict with other tickets.

**Is your feature request related to a problem? Please describe.**
I have a system that is running CFDP version 3.0 and want to interface it with a system that is running the older version of CFDP 2.2.  Are the two versions compatible for file transfers?

**Describe the solution you'd like**
I would like to know if CFDP version 2.2 and 3.0 are compatible for file transfers.

**Describe alternatives you've considered**
I have looked at the documentation for CFDP available on the github page and have not seen anything suggesting they are compatible/incompatible with each other.  

**Additional context**
I know certain things have changed across the two versions, such as the HK Telemetry messages having a different format.  
**Requester Info**
Jeff Anderson

**Comment by jphickey** [comment]
I don't know if anyone has specifically tested that setup but since they both conform to the same CCSDS-defined protocol, it should generally work.  The newer CF app does recognize some of the newer features in the CFDP protocol that the old version does not have, but that's mainly on receive to check that those features are NOT set (e.g it still doesn't do large files).

I'd recommend trying it and feeding any insights back.

I'd also recommend deploying the main line version as opposed to the previous v3.0.0 tag - there were a lot of fixes that went in to address troubles with the 3.0.0 version.  Should probably be a new tag coming sometime soon I think.

**Comment by space-individual** [comment]
Hi Joseph,
Thanks for the insight.  Is there any documentation that is available that lists the upgrade path from 2.2 to 3.0?
Thanks,
Jeff A.

**Comment by jhnphm** [comment]
Wouldn't this be an issue for instances of CF talking to one another? https://github.com/nasa/CF/issues/224
